crates/solver/src/liquidity/zeroex.rs

```diff
@@ -79,36 +120,136 @@ impl ZeroExLiquidity {
     }
 }
 
+/// Calls the 0x API to get all orders and initializes the cache with them.
+async fn init_cache(
+    api: Arc<dyn ZeroExApi>,
+    cache: Arc<Mutex<HashMap<String, OrderRecord>>>,
+    sender: Address,
+) -> Result<()> {
+    cache.lock().await.clear(); // init can be called multiple times, so clear the cache first
+
+    let queries = &[
+        // orders fillable by anyone
+        OrdersQuery::default(),
+        // orders fillable only by our settlement contract
+        OrdersQuery {
+            sender: Some(sender),
+            ..Default::default()
+        },
+    ];
+
+    let zeroex_orders_results =
+        futures::future::join_all(queries.iter().map(|query| api.get_orders(query))).await;
+    let mut zeroex_orders = vec![];
+    for result in zeroex_orders_results {
+        match result {
+            Ok(orders) => zeroex_orders.extend(orders),
+            Err(err) => {
+                return Err(anyhow::anyhow!(
+                    "ZeroExResponse error during liqudity fetching: {}",
+                    err
+                ));
+            }
+        }
+    }
```

Comment:
This should be equivalent
```suggestion
    let zeroex_orders =
        futures::future::try_join_all(queries.iter().map(|query| api.get_orders(query)))
            .await
            .context("failed to fetch 0x limit orders")?;
    let zeroex_orders: Vec<_> = zeroex_orders.into_iter().flatten().collect();
```

Reply:
Nice.